oci/private/config.bzl

```diff
@@ -0,0 +1,66 @@
+"repository rule that locates the .docker/config.json or containers/auth.json file."
+
+def _file_exists(rctx, path):
+    result = rctx.execute(["stat", path])
+    return result.return_code == 0
+
+# Path of the auth file is determined by the order described here;
+# https://github.com/google/go-containerregistry/tree/main/pkg/authn#tldr-for-consumers-of-this-package
+def _get_auth_file_path(rctx):
+    # this is the standard path where registry credentials are stored
+    config_path = "{}/.docker/config.json".format(rctx.os.environ["HOME"])
+
+    # set config path to DOCKER_CONFIG env if present
+    if "DOCKER_CONFIG" in rctx.os.environ:
+        config_path = rctx.os.environ["DOCKER_CONFIG"]
+
+    if _file_exists(rctx, config_path):
+        return config_path
+
+    # https://docs.podman.io/en/latest/markdown/podman-login.1.html#authfile-path
+    XDG_RUNTIME_DIR = "{}/.config".format(rctx.os.environ["HOME"])
+    if "XDG_RUNTIME_DIR" in rctx.os.environ:
+        XDG_RUNTIME_DIR = rctx.os.environ["XDG_RUNTIME_DIR"]
+
+    config_path = "{}/containers/auth.json".format(XDG_RUNTIME_DIR)
+
+    # podman support overriding the standard path for the auth file via this special environment variable.
+    # https://docs.podman.io/en/latest/markdown/podman-login.1.html#authfile-path
+    if "REGISTRY_AUTH_FILE" in rctx.os.environ:
+        config_path = rctx.os.environ["REGISTRY_AUTH_FILE"]
+
+    if _file_exists(rctx, config_path):
+        return config_path
+
+    return None
+
+def _oci_config_locator_impl(rctx):
+    config_path = _get_auth_file_path(rctx)
+    config_path = None
+    if not config_path:
+        # rctx.execute is cached between bazel invocations. prefer it over print
```

Comment:
the spam problem was in a single bazel invocation, so I don't think this is actually a reason to avoid print. We were simply evaluating that starlark statement too many times in a single invocation.

I guess `echo` is still fine, though it feels unusual

Reply:
output is nicer too

```WARNING: Ignoring JAVA_HOME, because it must point to a JDK, not a JRE.
Starting local Bazel server and connecting to it...
WARNING: Could not find the `$HOME/.docker/config.json` and `$XDG_RUNTIME_DIR/containers/auth.json` file. 
 Running one of `podman login`, `docker login`, `crane login` may help.
INFO: Analyzed 4 targets (51 packages loaded, 303 targets configured).
INFO: Found 4 targets...
INFO: Elapsed time: 22.658s, Critical Path: 2.06s
INFO: 7 processes: 5 internal, 2 local.
INFO: Build completed successfully, 7 total actions

```